examples/solids/qfunctions/hyperSS.h

```diff
@@ -319,5 +319,132 @@ CEED_QFUNCTION(HyperSSdF)(void *ctx, CeedInt Q, const CeedScalar *const *in,
+
+  // Quadrature Point Loop
+  CeedPragmaSIMD
+  for (CeedInt i=0; i<Q; i++) {
+    // Read spatial derivatives of u
+    // *INDENT-OFF*
+    const CeedScalar du[3][3]   = {{ug[0][0][i],
+                                    ug[1][0][i],
+                                    ug[2][0][i]},
+                                   {ug[0][1][i],
+                                    ug[1][1][i],
+                                    ug[2][1][i]},
+                                   {ug[0][2][i],
+                                    ug[1][2][i],
+                                    ug[2][2][i]}
+                                  };
+    // -- Qdata
+    const CeedScalar wJ         =   qdata[0][i];
+    const CeedScalar dXdx[3][3] = {{qdata[1][i],
+                                    qdata[2][i],
+                                    qdata[3][i]},
+                                   {qdata[4][i],
+                                    qdata[5][i],
+                                    qdata[6][i]},
+                                   {qdata[7][i],
+                                    qdata[8][i],
+                                    qdata[9][i]}
+                                  };
+    // *INDENT-ON*
+
+    // Compute gradu
+    //   dXdx = (dx/dX)^(-1)
+    // Apply dXdx to du = gradu
+    CeedScalar gradu[3][3];
+    for (int j = 0; j < 3; j++)     // Component
+      for (int k = 0; k < 3; k++) { // Derivative
+        gradu[j][k] = 0;
+        for (int m = 0; m < 3; m++)
+          gradu[j][k] += dXdx[m][k] * du[j][m];
+      }
+
+    // Compute Strain : e (epsilon)
+    // e = 1/2 (grad u + (grad u)^T)
+    const CeedScalar e00 = (gradu[0][0] + gradu[0][0])/2.,
+                     e01 = (gradu[0][1] + gradu[1][0])/2.,
+                     e02 = (gradu[0][2] + gradu[2][0])/2.,
+                     e11 = (gradu[1][1] + gradu[1][1])/2.,
+                     e12 = (gradu[1][2] + gradu[2][1])/2.,
+                     e22 = (gradu[2][2] + gradu[2][2])/2.;
+    // *INDENT-OFF*
+    const CeedScalar e[3][3] =  {{e00, e01, e02},
+                                 {e01, e11, e12},
+                                 {e02, e12, e22}
+                                };
+    // *INDENT-ON*
+
+    // strain (epsilon)
+    //    and
+    // stress (sigma) in Voigt notation:
+    //           [e00]              [sigma00]
+    //           [e11]              [sigma11]
+    // epsilon = [e22]  ,   sigma = [sigma22]
+    //           [e12]              [sigma12]
+    //           [e02]              [sigma02]
+    //           [e01]              [sigma01]
+    //
+    // mu = E / (2 * (1 + nu))
+    // bulk modulus = E / (2 * (1 - 2 * nu))
+    // lambda = (3 * bulk modulus - 2 * mu) / 3
+    // e_v = volumetric strain = e00 + e11 + e22
+    //
+    // sigma = lambda * log(1 + e_v) + 2 * mu * epsilon
+    //
+    // Above Voigt Notation is placed in a 3x3 matrix:
+    // Volumetric strain
+    const CeedScalar strain_vol = e[0][0] + e[1][1] + e[2][2];
```

Comment:
Thanks Jeremy. Do we want to add the calculation of the volumetric strain to the other material models as well? I think it could be a useful diagnostic variable.